### PR TITLE
21172 - removing intermediate step for keeping SIs in the UI

### PR DESCRIPTION
**Pull request by hfekete**
*Issue:*https://github.com/bcgov/entity/issues/

*Description of changes:*
- remove `SignitificantIndividualI `and replace it with `SiSchemaType`
- create converters for `SiSchemaType `to Bods schema 
- fix unit tests
- fix cypress tests
- some additional fixes and updates for summary table and significant individuals effective date
- update summary table to handle new schema type instead of interface

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-btr-dev--pr-163-r68zzvve.web.app

File: btr-web/btr-main-app/interfaces/control-of-directors-i.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-export interface ControlOfDirectorsI {
-    directControl: boolean
-    indirectControl: boolean
-    significantInfluence: boolean
-    inConcertControl: boolean
-}
+// export interface ControlOfDirectorsI {
```
Comment (kialj876) [inline btr-web/btr-main-app/interfaces/control-of-directors-i.ts:1]: can we remove the commented out interfaces? I don't think we need them anymore

File: btr-web/btr-main-app/services/file-significant-individual.ts [hunk 2]
```diff
@@ -109,6 +111,7 @@ const getPersonAndOwnershipAndControlStatements = (sif: SignificantIndividualFil
 }
 
 const convertToBtrBodsForSubmit = (sif: SignificantIndividualFilingI): BtrFilingI => {
+  // todo: fixme: with new changes
```
Comment (kialj876) [inline btr-web/btr-main-app/services/file-significant-individual.ts:114]: Can probably remove this comment or is there work needed in here for a future ticket?

File: btr-web/btr-main-app/utils/validation.ts [hunk 3]
```diff
@@ -72,12 +72,12 @@ export function validateBirthDate (birthDate: any): boolean {
...
 export function validateTaxNumberInfo (
-  taxData: { taxNumber: string | null, hasTaxNumber: boolean | null },
+  taxData: { taxNumber?: string, hasTaxNumber?: boolean },
   ctx: RefinementCtx
 ): never {
   const t = useNuxtApp().$i18n.t
-  if (taxData.hasTaxNumber) {
-    if (!checkSpecialCharacters(taxData.taxNumber)) {
+  if (taxData.hasTaxNumber && taxData.taxNumber) {
+    if (!checkSpecialCharacters(taxData.taxNumber!)) {
```
Comment (kialj876) [inline btr-web/btr-main-app/utils/validation.ts:80]: for this and below: is the extra '!' still necessary? You updated the typing of the function  to include 'undefined' as well so I don't think you need it right?

**Review by kialj876 — APPROVED** [review]
Nice work!

**Pull request merged**